nautobot/extras/models/roles.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from nautobot.extras.fields import LimitedChoiceField
+from nautobot.extras.models.base_properties import BasePropertiesModel
+from nautobot.extras.utils import RoleModelsQuery, extras_features
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class Role(BasePropertiesModel):
+    content_types = models.ManyToManyField(
+        to=ContentType,
+        related_name="roles",
+        verbose_name="Content type(s)",
+        limit_choices_to=RoleModelsQuery(),
+        help_text="The content type(s) to which this role applies.",
+    )
```

Comment:
Looking at the models this might eventually replace, `ipam.Role` also has a `weight` field, so we might consider including that on this model?